File: src/main/java/link/infra/packwiz/installer/ui/InputStateHandler.java
package link.infra.packwiz.installer.ui;

public class InputStateHandler {
	private boolean optionsButtonPressed = false;
	private boolean cancelButtonPressed = false;

	synchronized void pressCancelButton() {
		this.cancelButtonPressed = true;
	}

	synchronized void pressOptionsButton() {
		this.optionsButtonPressed = true;
	}

	public synchronized boolean getCancelButton() {
		return cancelButtonPressed;
	}

	public synchronized boolean getOptionsButton() {
		return optionsButtonPressed;
	}
}

File: src/main/kotlin/link/infra/packwiz/installer/ui/InputStateHandler.kt
package link.infra.packwiz.installer.ui

class InputStateHandler {
	// TODO: convert to coroutines/locks?
	@get:Synchronized
	var optionsButton = false
		private set
	@get:Synchronized
	var cancelButton = false
		private set

	@Synchronized
	fun pressCancelButton() {
		cancelButton = true
	}

	@Synchronized
	fun pressOptionsButton() {
		optionsButton = true
	}
}